images/ubuntu.yaml

```diff
@@ -666,16 +666,40 @@ actions:
   variants:
   - default
 
+- trigger: post-packages
+  action: |-
+    #!/bin/sh
+    set -eux
+
+    # Disable UA attach
+    systemctl mask ua-auto-attach
+
 - trigger: post-packages
   action: |-
     #!/bin/sh
     set -eux
 
     # Enable systemd-networkd
     systemctl enable systemd-networkd
+  releases:
+  - bionic
+  - focal
+  - jammy
+  - mantic
 
-    # Disable UA attach
-    systemctl mask ua-auto-attach
+  # Use NetworkManager from Noble release as systemd-networkd is causing
+  # delays on boot. Ubuntu Desktop images also come with NetworkManager by
+  # default, so we should start use it instead of systemd-networkd for
+  # all future releases.
+- trigger: post-packages
+  action: |-
+    #!/bin/sh
+    # Mask systemd-networkd
+    systemctl mask systemd-networkd
```

Comment:
In line 683 we enable it so I think we should instead just not enable it to begin with?

Reply:
Yes, but we enable it for all previous releases. I just wanted to ensure systemd-networkd is always disabled on Noble.

Reply:
Ah I see. However you made the line 683 apply only for Mantic and below so at worst it's redundant/safety?

Either way, please add a link to the LP bug https://bugs.launchpad.net/subiquity/+bug/2063331 just so that we remember to remove it once fixed.

Reply:
I just checked and it's not enabled on Noble if we don't do it ourselves. So let's not mask it then, if that's OK for you.

```
root@nd1:~# systemctl is-enabled systemd-networkd
disabled
```